frontend/packages/console-app/src/components/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,279 @@
+import {
+  PersistentVolumeClaimModel,
+  VolumeSnapshotModel,
+  VolumeSnapshotClassModel,
+  StorageClassModel,
+} from '@console/internal/models';
+import { PVCDropdown } from '@console/internal/components/utils/pvc-dropdown';
+
+import './_create-volume-snapshot.scss';
+import { Status } from '@console/shared/src/components/status';
+import { fetchK8s } from '@console/internal/graphql/client';
+
+const SnapshotClassDropdown: React.FC<SnapshotClassDropdownProps> = (props) => {
+  const kind = referenceForModel(VolumeSnapshotClassModel);
+  const { selectedKey, pvcProvisioner } = props;
+  const resources = [{ kind }];
+  const provisionerFilter = (snapshotClass: VolumeSnapshotClassKind) => {
+    return pvcProvisioner && pvcProvisioner.includes(snapshotClass?.driver);
+  };
+
+  return (
+    <ListDropdown
+      {...props}
+      desc="Volume Snapshot Class with same provisioner as claim"
+      dataFilter={provisionerFilter}
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select snapshot class"
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const PVCSummary: React.FC<PVCSummaryProps> = ({
+  namespace,
+  pvc,
+  status,
+  persistentVolumeClaim,
+}) => {
+  const storageClass = persistentVolumeClaim?.spec?.storageClassName || '-';
+  const requestedCapacity = persistentVolumeClaim?.status?.capacity?.storage || '-';
+  const accessModes =
+    persistentVolumeClaim?.spec?.accessModes.reduce((allModes: string, mode: string) => {
+      return `${mode} ${allModes}`;
+    }, '') || '-';
+  const volumeMode = persistentVolumeClaim?.spec?.volumeMode || '-';
+  return (
+    <dl>
+      <dt className="co-volume-snapshot__details-body">Persistent Volume Claim Details</dt>
+      <dt>Name</dt>
+      <dd>
+        <ResourceIcon kind="PersistentVolumeClaim" />
+        {pvc}
+      </dd>
+      <dt>Namespace</dt>
+      <dd>
+        <ResourceIcon kind="Namespace" />
+        {namespace}
+      </dd>
+      <dt>Status</dt>
+      <dd>
+        <Status status={status} />
+      </dd>
+      <dt>Storage Class</dt>
+      <dd>
+        <ResourceIcon kind="StorageClass" />
+        {storageClass}
+      </dd>
+      <dt>Requested Capacity</dt>
+      <dd>{requestedCapacity}</dd>
+      <dt>Access Mode</dt>
+      <dd>{accessModes}</dd>
+      <dt>Volume Mode</dt>
+      <dd>{volumeMode}</dd>
+    </dl>
+  );
+};
+
+const CreateSnapshotForm: React.FC<SnapshotResourceProps> = (props) => {
+  const { resourceName, namespace, kindObj } = props;
+
+  const [pvcName, setPVCName] = React.useState(resourceName || '');
+  const [pvcObj, setPVCObj] = React.useState(null);
+  const [snapshotName, setSnapshotName] = React.useState<string>(`${pvcName}-snapshot`);
+  const [snapshotClassName, setSnapshotClassName] = React.useState<string>();
+  const [error, setError] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+  const [pvcStatus, setPVCStatus] = React.useState<string>(null);
```

Comment:
Shouldn't this be initialized with empty string instead of null here? I don't see that the value `null` is being used anywhere? Same for `pvcProvisioner`